src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountDecider.fs

```diff
@@ -0,0 +1,80 @@
+﻿module Deciders.BankAccountDecider
+
+open System
+open BankAccount
+open BankAccountPrimitives
+
+type OpenBankAccount =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      Now: DateTimeOffset }
+
+type RecordDeposit = { Amount: decimal; CashierId: CashierId; Now: DateTimeOffset }
+type WithdrawCashFromATM = { Amount: decimal; AtmId: AtmId; Now: DateTimeOffset }
+type CloseBankAccount = { Reason: string; Now: DateTimeOffset }
+
+type Command =
+    | OpenBankAccount of OpenBankAccount
+    | RecordDeposit of RecordDeposit
+    | WithdrawCashFromATM of WithdrawCashFromATM
+    | CloseBankAccount of CloseBankAccount
+
+let openBankAccount (command: OpenBankAccount) (bankAccount: BankAccount) =
+    match bankAccount with
+    | Open _ -> invalidOp "Account is already opened!"
+    | Closed _ -> invalidOp "Account is already closed!"
+    | NotInitialised _ ->
+        BankAccountOpened
+            { BankAccountId = command.BankAccountId
+              AccountNumber = command.AccountNumber
+              ClientId = command.ClientId
+              CurrencyISOCode = command.CurrencyISOCode
+              CreatedAt = command.Now
+              Version = 1 }
+
+let recordDeposit (command: RecordDeposit) (bankAccount: BankAccount) =
+    match bankAccount with
+    | NotInitialised _ -> invalidOp "Account is not opened!"
+    | Closed _ -> invalidOp "Account is closed!"
+    | Open openBankAccount ->
+        DepositRecorded
+            { BankAccountId = openBankAccount.Id
+              Amount = command.Amount
+              CashierId = command.CashierId
+              RecordedAt  = command.Now
+              Version = openBankAccount.Version + 1L }
+
+let withdrawCashFromATM (command: WithdrawCashFromATM) (bankAccount: BankAccount) =
+    match bankAccount with
+    | NotInitialised _ -> invalidOp "Account is not opened!"
+    | Closed _ -> invalidOp "Account is closed!"
+    | Open openBankAccount ->
+        if (openBankAccount.Balance < command.Amount) then
+            invalidOp "Not enough money!"
+
+        CashWithdrawnFromATM
+            { BankAccountId = openBankAccount.Id
+              Amount = command.Amount
+              ATMId = command.AtmId
+              RecordedAt = command.Now
+              Version = openBankAccount.Version + 1L }
+
+let closeBankAccount (command: CloseBankAccount) (bankAccount: BankAccount) =
+    match bankAccount with
+    | NotInitialised _ -> invalidOp "Account is not opened!"
+    | Closed _ -> invalidOp "Account is already closed!"
+    | Open openBankAccount ->
+        BankAccountClosed
+            { BankAccountId = openBankAccount.Id
+              Reason = command.Reason
+              ClosedAt = command.Now
+              Version = openBankAccount.Version + 1L }
+
+let decide command bankAccount =
```

Comment:
I'd say that it's just the `state` of the thing (we're in module BankAccountDecider)

Reply:
same justification as above https://github.com/oskardudycz/Ogooreck/pull/13#discussion_r1002662002

Reply:
Same counterargument :D

Reply:
let's agree to disagree then :P I might change my mind in the future :smile: